### PR TITLE
[Misc] update fp8 to use `vLLMParameter`

### PR DESCRIPTION
**Pull request by dsikka**
# Summary
- Update fp8 to use vLLMParameters

### CODE REVIEW

File: vllm/model_executor/layers/linear.py [hunk 1]
```diff
@@ -1005,6 +1005,12 @@ def weight_loader(self, param: Parameter, loaded_weight: torch.Tensor):
 
     def weight_loader_v2(self, param: BasevLLMParameter,
                          loaded_weight: torch.Tensor):
+
+        # Special case for loading scales off disk, which often do not
+        # have a shape (such as in the case of AutoFP8).
+        if len(loaded_weight.shape) == 0:
```
Comment (mgoin) [inline vllm/model_executor/layers/linear.py:1011]: can you check that `loaded_weight.numel() == 1` as well?

File: vllm/model_executor/parameter.py [hunk 2]
```diff
@@ -208,10 +208,17 @@ def _shard_id_as_int(self, shard_id: Union[str, int]) -> int:
         if isinstance(shard_id, int):
             return shard_id
 
+        # if not int, assuem shard_id for qkv
```
Comment (mgoin) [inline vllm/model_executor/parameter.py:211]:
```suggestion
        # if not int, assume shard_id for qkv
```

File: vllm/model_executor/layers/quantization/fp8.py [hunk 3]
```diff
@@ -137,6 +138,7 @@ def create_weights(
     ):
         del input_size, output_size
         output_size_per_partition = sum(output_partition_sizes)
+        weight_loader = extra_weight_attrs.get("weight_loader")
```
Comment (mgoin) [inline vllm/model_executor/layers/quantization/fp8.py:141]: Are there cases where we aren't guaranteed to have a weight_loader here?

**Review by mgoin — APPROVED** [review]
Straightforward, looks good to me! I'm not sure if the failing AMD tests are directly related to this

File: vllm/model_executor/layers/quantization/fp8.py (hunk 3 above)
Comment (dsikka) [reply vllm/model_executor/layers/quantization/fp8.py:141]: I dont think so - unless we made changes to the `LinearBase` lifecycle

**Review by mgoin — APPROVED** [review]

**Pull request merged**